src/rest-server/src/models/k8s-secret.js

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const axios = require('axios');
+const _ = require('lodash');
+const {Agent} = require('https');
+const {URL} = require('url');
+
+const {apiserver} = require('@pai/config/kubernetes');
+
+const initClient = (namespace) => {
+  if (!namespace) {
+    throw new Error('K8S SECRET: invalid namespace');
+  }
+  const config = {
+    baseURL: new URL(`/api/v1/namespaces/${namespace}/secrets`, apiserver.uri).href,
+    maxRedirects: 0,
+    headers: {
+      'Accept': 'application/json',
+    },
+  };
+  if (apiserver.ca) {
+    config.httpsAgent = new Agent({ca: apiserver.ca});
+  }
+  if (apiserver.token) {
+    config.headers['Authorization'] = `Bearer ${apiserver.token}`;
+  }
+  return axios.create(config);
+};
+
+const serialize = (object) => {
+  if (object == null) {
+    return {};
+  }
+
+  const result = {};
+  for (const [key, val] of Object.entries(object)) {
+    result[key] = Buffer.from(val).toString('base64');
+  }
+  return result;
+};
+
+
+const deserialize = (object) => {
+  if (object == null) {
+    return {};
+  }
+
+  const result = {};
+  for (const [key, val] of Object.entries(object)) {
+    result[key] = Buffer.from(val, 'base64').toString();
+  }
+  return result;
+};
+
+const encodeLabeLSelector = (labelSelector) => {
+  const builder = [];
+  for (const [key, val] of Object.entries(labelSelector)) {
+    builder.push(`${key}=${val}`);
+  }
+  return builder.join(',');
+};
+
+const createNamespace = async (namespace) => {
+  const url = new URL(`/api/v1/namespaces/`, apiserver.uri).href;
+  const config = {
```

Comment:
Is it deplicate with initClient

Reply:
It's a namespace api with different base uri.
Maybe it should be in another file like `k8s-util.js` or `k8s-namespace.js`.
I think we can reorganize the k8s api file structure later.